src/abstractdatatable/abstractdatatable.jl

```diff
@@ -696,24 +667,21 @@ Base.hcat(dt::AbstractDataTable, x, y...) = hcat!(hcat(dt, x), y...)
 Base.hcat(dt1::AbstractDataTable, dt2::AbstractDataTable, dtn::AbstractDataTable...) = hcat!(hcat(dt1, dt2), dtn...)
 
 @generated function promote_col_type(cols::AbstractVector...)
-    elty = Base.promote_eltype(cols...)
-    if elty <: Nullable
-        elty = eltype(elty)
-    end
-    if elty <: CategoricalValue
-        elty = elty.parameters[1]
+    T = promote_type(map(x-> eltype(x) >: Null ? Nulls.T(eltype(x)) : eltype(x), cols)...)
```

Comment:
Not sure whether we should add that function, but it's easy to obtain this behavior using `Nulls.T(Union{elty, Null})`.